docs/nodemcu-lrm.md

```diff
@@ -0,0 +1,331 @@
+
+#### luaL_posttask
+
+`  int luaL_posttask (lua_State* L, int prio);`         [-1, +0, e]
+
+Posts a task to execute the function popped from the stack at the specified user task priority
+-  `prio` one of:
+   -  `LUA_TASK_LOW`
+   -  `LUA_TASK_MEDIUM`
+   -  `LUA_TASK_HIGH`
+
+Note that the function is invoked with the priority as its parameter.
+
+#### luaL_pushlfsmodule
+
+`  int luaL_pushlfsmodule ((lua_State *L);`         [-1, +1, -]
+
+This function pops a module name from the stack. If this is a string, LFS is loaded and it contains the named module then its closure is pushed onto the stack as a function value, otherwise `nil` is pushed.
+
+Returns the type of the pushed value.
+
+#### luaL_pushlfsmodules
+
+`  int luaL_pushlfsmodules (lua_State *L);`         [-0, +1, m]
+
+If LFS is loaded then an array of the names of all of the modules in LFS is pushed onto the stack as a function value, otherwise `nil` is pushed.
+
+Returns the type of the pushed value.
+
+#### luaL_pushlfsdts
+
+`  int luaL_pushlfsdts (lua_State *L);`         [-0, +1, m-]
+
+If LFS is loaded then the Unix-style date-timestamp for the compile time of the image is pushed onto the stack as a integer value, otherwise `nil` is pushed.  Note that the primary use of this stamp is to act as a unique identifier for the image version.
+
+Returns the type of the pushed value.
+
+#### luaL_reref
+`  void (luaL_reref) (lua_State *L, int t, int *ref);`         [-1, +0, m]
+
+Variant of [luaL_ref](https://www.lua.org/manual/5.3/manual.html#luaL_ref).  If `*ref` is a valid reference in the table at index t, then this is replaced by the object at the top of the stack (and pops the object), otherwise it creates and returns a new reference using the `luaL_ref` algorithm.
+
+#### luaL_rometatable
+
+`  int (luaL_rometatable) (lua_State *L, const char* tname, const ROTable *p);`         [-0, +1, e]
+
+Equivalent to `luaL_newmetatable()` for ROTable metatables.  Adds key / ROTable entry to the registry  `[tname] = p`, rather than using a new RAM table.
+
+#### luaL_unref2
+
+`  luaL_unref2(l,t,r)`
+
+This macro executes `luaL_unref(L, t, r)` and then assigns `r = LUA_NOREF`.
+
+### Declaring modules and ROTables in NodeMCU
+
+All NodeMCU C library modules should include the standard header "`module.h`".  This internally includes `lnodemcu.h` and these together provide the macros to enable declaration of NodeMCU modules and ROTables within them.  All ROtable support macros are wither prefixed by `LRO_` (Lua Read Only) or in the case of table entries `LROT_`. 
+
+#### NODEMCU_MODULE
+
+` NODEMCU_MODULE(sectionname, libraryname, map, initfunc)`
+
+This macro enables the module to be statically declared and linked in the `ROM` ROTable.  The global environment's metafield `__index=ROM` hence any entries in the ROM table are resolved as read-only entries in the global environment.
```

Comment:
"metafield" -> "metafield is set as" or some such? It took me a couple of reads to grok the meaning of the sentence as is.